src/components/Pages/Submit/Form/store/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { useImmer } from "use-immer";
+import { ArticleParams, ArticleStore, Tag } from "./types";
+
+export const useArticleStore = (articleProps?: ArticleStore) => {
+  const blank = {
+    title: "",
+    abstract: "",
+    article: "",
+    content: "",
+    paperName: "",
+    authors: [],
+    permalink: "",
+    submitName: "",
+    submitLink: "",
+    submissionBy: { name: "", link: "" },
+    doi: "",
+    references: [],
+    notes: [],
+    links: [{ link_title: "", link_url: "" }],
+    tags: [] as string[],
+  };
+  const initialValue = articleProps || blank;
+  const [article, setArticle] = useImmer<ArticleStore>(initialValue);
+  const [oldPermalink, setOldPermalink] = useState<string>();
+
+  useEffect(() => {
+    if (article.permalink !== "") {
+      localStorage.setItem(article.permalink, JSON.stringify(article));
+    }
+  }, [article]);
+
+  useEffect(() => {
+    if (oldPermalink) {
+      localStorage.removeItem(oldPermalink);
+      setOldPermalink(undefined);
+    }
+  }, [oldPermalink]);
+
+  const upsert = useCallback((update: ArticleParams) => {
```

Comment:
So there's a lot going on in this function

Why useCallback

basically react is incredibly dumb and will rerender the function each time this hook is called which can cause massive issues if you are sending it as a prop. To deal with this we use this callback which basically creates a state that is listened for any changes in the []. It's a simple function so we don't need it to change forms based on any props to the hook itself. Yes I know it's so much bloat, why is react this way ectra. 

[docs](https://react.dev/reference/react/useCallback
)

use-immer is basically just an Idempotent helper. Your structure is simple enough that it's overkill at the moment but it helped me make sure nothing got lost with my minimal understanding of the code. 

So let's break down the function itself. I wanted something that could take any parameter and just work, be it multiple parts or how it is right now with one attribute per update. 

This is what the ArticleParam type is allowing. Everything is optional

on Line 42 I deconstruct the props between ones that have more logic and ones we can just replace directly. 
I then loop through each of the latter and replace. There is probably an easier way of mass updating but that would requiring a construction of fully new object. Which causes immer to fail

then we check that each parameter exists and if it does we do the change that it requires. Which is either additional string manipulation or not a one to one placement on the store object